<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
<script>
//网络请求：aaa -> 自己处理10行
//处理：aaa111 -> 自己处理10行
//处理：aaa111222 -> 自己处理
// new Promise((resolve, reject)=>{
//     setTimeout(()=>{
//         resolve('aaa')
//     },1000)
// }).then(res=>{
//     //自己处理10行代码
//     console.log(res,'第一层的10行处理代码')
//
//     //对结果进行第一次处理
//     return new Promise((resolve)=>{
//         resolve(res,+'111')
//     })
// }).then(res=>{
//     console.log(res,'第二层的10行处理代码')
//
//     return new Promise(resolve=>{
//         resolve(res+'222')
//     })
// }).then(res=>{
//     console.log(res,"第三层的10行处理代码")
// })

//error 的简写
new Promise((resolve, reject)=>{
    setTimeout(()=>{
        resolve('aaa')
    },1000)
}).then(res=>{
    //自己处理10行代码
    console.log(res,'第一层的10行处理代码')

    //对结果进行第一次处理
    // return Promise.reject('error message')
    throw 'error message'
}).then(res=>{
    console.log(res,'第二层的10行处理代码')

    return Promise.resolve(res+'222')
}).then(res=>{
    console.log(res,"第三层的10行处理代码")
}).catch(err=>{
    console.log(err)
})

//更加简洁的写法--省略掉Promise.resolve（内部会自己调用Promise.resolve）
new Promise((resolve, reject)=>{
    setTimeout(()=>{
        resolve('aaa')
    },1000)
}).then(res=>{
    //自己处理10行代码
    console.log(res,'第一层的10行处理代码')

    //对结果进行第一次处理
    return res+'111'
}).then(res=>{
    console.log(res,'第二层的10行处理代码')

    return res+'222'
}).then(res=>{
    console.log(res,"第三层的10行处理代码")
})

</script>
</body>
</html>